internal/util.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import "encoding/binary"
+
+// Finds the byte in the byte array bs
+// at offset offset, and determines whether it is 1 or 0.
+// return true if the nth bit is 1, false otherwise.
+// from MSB to LSB order
+func GetNthBit(bs []byte, offset int) bool {
+	arrayOffset := offset / 8
+	bitOfByte := offset % 8
+
+	masked := int(bs[arrayOffset] & (1 << uint(7-bitOfByte)))
+	return masked != 0
+}
+
+func LongToBytes(num int64) []byte {
+	buf := make([]byte, 8)
+	binary.LittleEndian.PutUint64(buf, uint64(num))
+	return buf
+}
+
+func IntToBytes(num int) []byte {
+	buf := make([]byte, 4)
+	binary.LittleEndian.PutUint32(buf, uint32(num))
+	return buf
+}
+
+// In each byte, the bits are ordered MSB to LSB
+func ToBytes(bits []bool) []byte {
+	bs := make([]byte, (len(bits)+7)/8)
+	for i := 0; i < len(bits); i++ {
+		if bits[i] {
+			bs[i/8] |= (1 << 7) >> uint(i%8)
+		}
+	}
+	return bs
+}
```

Comment:
Yup, I did write some tests for those methods but after all those tests passed, I deleted them all. My bad :(